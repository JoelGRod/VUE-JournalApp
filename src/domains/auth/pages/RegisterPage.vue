<template>
  <span class="login100-form-title p-b-41"> Register </span>
  <form @submit.prevent="onSubmit" class="login100-form validate-form p-b-33 p-t-5">

    <div class="wrap-input100 validate-input" data-validate="Enter username">
      <input v-model="registerForm.name" class="input100" type="text" placeholder="Username" required />
      <span class="focus-input100" data-placeholder="&#xe82a;"></span>
    </div>

    <div class="wrap-input100 validate-input" data-validate="Enter email">
      <input v-model="registerForm.email" class="input100" type="email" placeholder="Email" required />
      <span class="focus-input100" data-placeholder="&#xe818;"></span>
    </div>

    <div class="wrap-input100 validate-input" data-validate="Enter password">
      <input
        v-model="registerForm.password"
        class="input100"
        type="password"
        placeholder="Password"
        required
      />
      <span class="focus-input100" data-placeholder="&#xe80f;"></span>
    </div>

    <div class="container-login100-form-btn m-t-32">
      <button class="login100-form-btn" type="submit">Register</button>
    </div>

    <div class="container-login100-form-btn m-t-32">
      <router-link :to="{ name: 'Login' }">Login</router-link>
    </div>
  </form>
</template>

<script>
import { ref } from "vue"
import useAuth from "../composables/useAuth";

export default {

  setup() {

    const { createUser } = useAuth()

    const registerForm = ref({
      name: 'test',
      email: 'test@test.com',
      password: '123456'
    })

    return {
      // Public Properties
      registerForm,
      // Public methods
      onSubmit: async () => {
        const resp = await createUser( registerForm.value )
        console.log(resp)
      }
    }

  }
};
</script>