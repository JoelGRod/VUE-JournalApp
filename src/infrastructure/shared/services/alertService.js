import Swal from 'sweetalert2'

export const showLoader = () => {
    new Swal({
        title: 'Please Wait...',
        allowOutsideClick: false
    })
    Swal.showLoading()
}

export const showSuccess = (title, text) => {
    Swal.fire(title, text, 'success')
}

export const showError = (title, text) => {
    Swal.fire(title, text, 'error')
}

export const showDesicion = async ( title, text ) => {
    const { isConfirmed } = await Swal.fire({
        title,
        text,
        showDenyButton: true,
        confirmButtonText: 'Yes, i\'m sure'
    })
    return isConfirmed
}