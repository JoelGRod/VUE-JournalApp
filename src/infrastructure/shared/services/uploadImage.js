import cloudinaryUploadApi from "../api/cloudinaryApi";

const uploadImage = async( file ) => {
    if( !file ) return

    try {
        // Prepare body
        const formData = new FormData()
        formData.append('upload_preset', 'vue-tests')
        formData.append('file', file)
        // Send request
        const { data } = await cloudinaryUploadApi.post('', formData)
        // Return specific data
        return data.secure_url
    } catch (error) {
        console.log('Unable to upload image, call administrator')
        console.log(error)
        return null
    }
}

export default uploadImage