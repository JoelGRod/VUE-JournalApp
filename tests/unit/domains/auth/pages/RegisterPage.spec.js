import { shallowMount } from "@vue/test-utils";

import createVuexStore from "../../../mock-data/mock-store";
import RegisterPage from "@/domains/auth/pages/RegisterPage";

/* ----------- Alert Service Mock ----------- */
// Third party library mock
// import Swal from 'sweetalert2';
// module mock
import * as swal from "@/infrastructure/shared/services/alertService";
// module mock
jest.mock("@/infrastructure/shared/services/alertService", () => ({
  showLoader: jest.fn(),
  showSuccess: jest.fn(),
  showError: jest.fn(),
  showDecision: jest.fn(),
}));
/* ----------- Alert Service Mock ----------- */

/* ----------- Router Mock Composition API ----------- */
import {
  VueRouterMock,
  createRouterMock,
  injectRouterMock,
} from "vue-router-mock";

import { config } from "@vue/test-utils";

// create one router per test file
const router = createRouterMock();
// eslint-disable-next-line no-undef
beforeEach(() => {
  injectRouterMock(router);
});

// Add properties to the wrapper
config.plugins.VueWrapper.install(VueRouterMock);
/* ----------- Router Mock Composition API ----------- */

describe("Testing: RegisterPage.vue", () => {
  const store = createVuexStore({
    status: "not-authenticated",
    user: null,
    idToken: null,
    refreshToken: null,
  });

  store.dispatch = jest.fn();

  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(RegisterPage, {
      global: {
        plugins: [store],
      },
    });
  });

  test("should match with the snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("Should show error if email exists", async () => {
    // Arrange
    store.dispatch.mockReturnValueOnce({ ok: false, msg: "EMAIL_EXISTS" });
    // Act
    await wrapper.find("form.p-t-5").trigger("submit");
    // Assert
    expect(store.dispatch).toHaveBeenCalledWith("auth/createUser", {
      name: "",
      email: "",
      password: "",
    });
    expect(swal.showError).toHaveBeenCalledWith(
      "Something has not gone well",
      "EMAIL_EXISTS"
    );
  });

  test("should register user and redirect to daybook module", async () => {
    // Arrange
    store.dispatch.mockReturnValueOnce({ ok: true });
    const [name, email, password] = wrapper.findAll("input");
    // Act
    await name.setValue("test");
    await email.setValue("test@email.com");
    await password.setValue("123456");
    await wrapper.find("form").trigger("submit");
    // Assert
    expect(store.dispatch).toHaveBeenCalledWith("auth/createUser", {
      name: "test",
      email: "test@email.com",
      password: "123456",
    });
    expect(swal.showSuccess).toHaveBeenCalledWith(
      "Congratulations!",
      "The user has been created"
    );
    expect(wrapper.router.push).toHaveBeenCalledWith({
      name: "Daybook-No-Entry",
    });
  });
});
