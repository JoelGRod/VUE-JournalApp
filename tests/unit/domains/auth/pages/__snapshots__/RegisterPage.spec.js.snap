// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing: RegisterPage.vue should match with the snapshot 1`] = `
<span class="login100-form-title p-b-41"> Register </span>
<form class="login100-form validate-form p-b-33 p-t-5">
  <div class="wrap-input100 validate-input" data-validate="Enter username"><input class="input100" type="text" placeholder="Username" autocomplete="username" required=""><span class="focus-input100" data-placeholder=""></span></div>
  <div class="wrap-input100 validate-input" data-validate="Enter email"><input class="input100" type="email" placeholder="Email" autocomplete="username" required=""><span class="focus-input100" data-placeholder=""></span></div>
  <div class="wrap-input100 validate-input" data-validate="Enter password"><input class="input100" type="password" placeholder="Password" autocomplete="new-password" required=""><span class="focus-input100" data-placeholder=""></span></div>
  <div class="container-login100-form-btn m-t-32"><button class="login100-form-btn" type="submit">Register</button></div>
  <div class="container-login100-form-btn m-t-32">
    <router-link-stub to="[object Object]" replace="false" custom="false" ariacurrentvalue="page"></router-link-stub>
  </div>
</form>
`;
